Index: app/src/main/java/services/BackendAuthenticationService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/services/BackendAuthenticationService.java	(date 1605349425000)
+++ app/src/main/java/services/BackendAuthenticationService.java	(date 1605429391191)
@@ -27,10 +27,10 @@
     public static final String INTENT_KEY_FOR_RESULT_RECEIVER = "result receiver listener";
     private static final String BUNDLE_KEY_BACKEND_TO_FRONTEND_DECLARATION_OF_AUTHENTICATION_RESULT = "on authentication result issued";
 
-    private static final String BUNDLE_KEY_BACKEND_TO_FRONTEND_POST_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT = "frontend reply to backend on authentication result issued";
-    private static final String MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_SUCCESSFUL = "onSuccessful";
-    private static final String MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_FAILED = "onFailed";
-    private static final String MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_ERROR = "onError";
+    private static final String BUNDLE_KEY_BACKEND_TO_FRONTEND_ON_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT_RECEIVED = "frontend reply to backend on authentication result issued";
+    private static final String MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_SUCCESSFUL = "onSuccessful";
+    private static final String MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_FAILED = "onFailed";
+    private static final String MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_ERROR = "onError";
 
     public static final String BUNDLE_KEY_ON_RESULT_RECEIVER_SEND = "onResultReceiver sends";
     public static final String BUNDLE_VALUE_ON_RESULT_RECEIVER_SEND_ON_AUTHENTICATION_ERROR = "onResultReceiver sends call onError";
@@ -198,25 +198,25 @@
                         msg.sendToTarget();
                         break;
 
-                    case BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_ERROR:
-                        Log.w(BackendAuthenticationService.TAG + "." + this.TAG, "call" + " " + BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_ERROR);
-                        bundle.putString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_POST_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT,
+                    case BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_ERROR:
+                        Log.w(BackendAuthenticationService.TAG + "." + this.TAG, "call" + " " + BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_ERROR);
+                        bundle.putString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_ON_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT_RECEIVED,
                                 BackendAuthenticationService.BUNDLE_VALUE_ON_AUTHENTICATION_ERROR);
                         msg.setData(bundle);
                         msg.sendToTarget();
                         break;
 
-                    case BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_SUCCESSFUL:
-                        Log.w(BackendAuthenticationService.TAG + "." + this.TAG, "call" + " " + BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_SUCCESSFUL);
-                        bundle.putString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_POST_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT,
+                    case BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_SUCCESSFUL:
+                        Log.w(BackendAuthenticationService.TAG + "." + this.TAG, "call" + " " + BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_SUCCESSFUL);
+                        bundle.putString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_ON_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT_RECEIVED,
                                 BackendAuthenticationService.BUNDLE_VALUE_ON_AUTHENTICATION_SUCCESSFUL);
                         msg.setData(bundle);
                         msg.sendToTarget();
                         break;
 
-                    case BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_FAILED:
-                        Log.w(BackendAuthenticationService.TAG + "." + this.TAG, "call" + " " + MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_FAILED);
-                        bundle.putString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_POST_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT,
+                    case BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_FAILED:
+                        Log.w(BackendAuthenticationService.TAG + "." + this.TAG, "call" + " " + MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_FAILED);
+                        bundle.putString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_ON_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT_RECEIVED,
                                 BackendAuthenticationService.BUNDLE_VALUE_ON_AUTHENTICATION_FAILED);
                         msg.setData(bundle);
                         msg.sendToTarget();
@@ -263,7 +263,7 @@
         public boolean handleMessage(@NonNull Message message) {
             Bundle data = message.getData();
             boolean op1 = data.containsKey(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_DECLARATION_OF_AUTHENTICATION_RESULT);
-            boolean op2 = data.containsKey(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_POST_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT);
+            boolean op2 = data.containsKey(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_ON_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT_RECEIVED);
             Message msg = new Message();
             msg.setTarget(mBackendHandlerThread.getInstanceOfConsumerHandler());
             Bundle bundle = new Bundle();
@@ -273,29 +273,29 @@
                 switch (backendAuthenticationResult) {
                     case AUTHENTICATION_RESULT_IS_ERROR:
                         Log.i(BackendAuthenticationService.TAG + "." + this.TAG, "AUTHENTICATION ERROR");
-                        msg.obj = BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_ERROR;
-                        bundle.putString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_POST_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT,
-                                BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_ERROR);
+                        msg.obj = BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_ERROR;
+                        bundle.putString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_ON_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT_RECEIVED,
+                                BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_ERROR);
                         break;
 
                     case AUTHENTICATION_RESULT_IS_SUCCESSFUL:
                         Log.i(BackendAuthenticationService.TAG + "." + this.TAG, "AUTHENTICATION SUCCESSFUL");
-                        msg.obj = BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_SUCCESSFUL;
-                        bundle.putString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_POST_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT,
-                                BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_SUCCESSFUL);
+                        msg.obj = BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_SUCCESSFUL;
+                        bundle.putString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_ON_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT_RECEIVED,
+                                BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_SUCCESSFUL);
                         break;
 
                     case AUTHENTICATION_RESULT_IS_FAILED:
                         Log.i(BackendAuthenticationService.TAG + "." + this.TAG, "AUTHENTICATION FAILED");
-                        msg.obj = BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_FAILED;
-                        bundle.putString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_POST_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT,
-                                BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_FOR_AUTHENTICATION_RESULT_IS_FAILED);
+                        msg.obj = BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_FAILED;
+                        bundle.putString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_ON_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT_RECEIVED,
+                                BackendAuthenticationService.MSG_TITLE_FRONTEND_TO_BACKEND_ACKNOWLEDGEMENT_OF_AUTHENTICATION_RESULT_IS_FAILED);
                         break;
 
                     default:
                 }
             } else if (op2) {
-                String res = data.getString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_POST_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT);
+                String res = data.getString(BackendAuthenticationService.BUNDLE_KEY_BACKEND_TO_FRONTEND_ON_ACKNOWLEDGMENT_OF_AUTHENTICATION_RESULT_RECEIVED);
                 Bundle bundleResRec = null;
                 switch (res) {
                     case BackendAuthenticationService.BUNDLE_VALUE_ON_AUTHENTICATION_ERROR:
@@ -322,14 +322,6 @@
                         mResultReceiverListener.send(Activity.RESULT_OK, bundleResRec);
                         break;
 
-                    /*case BackendAuthenticationService.BUNDLE_VALUE_BACKEND_TO_FRONTEND_CALL_ON_AUTHENTICATION_UNKNOWN:
-                        Log.i(BackendAuthenticationService.TAG + "." + this.TAG, BUNDLE_VALUE_BACKEND_TO_FRONTEND_CALL_ON_AUTHENTICATION_UNKNOWN);
-                        bundleResRec = new Bundle();
-                        bundleResRec.putString(BackendAuthenticationService.BUNDLE_KEY_ON_RESULT_RECEIVER_SEND,
-                                BackendAuthenticationService.BUNDLE_VALUE_ON_RESULT_RECEIVER_SEND_ON_AUTHENTICATION_UNKNOWN);
-                        mResultReceiverListener.send(Activity.RESULT_OK, bundleResRec);
-                        break;*/
-
                     default:
                         bundleResRec = new Bundle();
                         bundleResRec.putString(BackendAuthenticationService.BUNDLE_KEY_ON_RESULT_RECEIVER_SEND,
